platform/settings.cpp

```diff
@@ -425,18 +426,16 @@ void UsageStats::EnterBackground()
   m_ss.SetValue(m_sessions, ToString(m_sessionsCount));
 }
 
-}  // namespace settings
-
-/*
-namespace marketing
-{
-Settings::Settings() : platform::StringStorageBase(GetPlatform().SettingsPathForFile(MARKETING_SETTINGS_FILE_NAME)) {}
-
-// static
-Settings & Settings::Instance()
+bool UsageStats::IsLoyalUser() const
 {
-  static Settings instance;
-  return instance;
+  #ifdef DEBUG
+  uint32_t constexpr kMinTotalForegroundTimeout = 30;
+  uint32_t constexpr kMinSessionsCount = 3;
+  #else
+  uint32_t constexpr kMinTotalForegroundTimeout = 60 * 15; // 15 min
```

Comment:
Could we make a longer lead-in delay for brand new users? As proposed in https://github.com/organicmaps/organicmaps/pull/8638#discussion_r1846961094

> Its better to allow much longer time (e.g. a month) before asking new users to donate. Allow them time to use the app in a real situation (e.g. a trip), learn its functionality and understand/appreciate value it brings.
> Also it'd help in cases when users re-install the app often. Alpha/beta testers included?

E.g. if its the first app start then delay the popup for a month?

Reply:
The 15 in foreground is about 2-3 days of average usage. @vng has made a test somewhere in comments...

Reply:
15 minutes is very little in terms of getting to know the app.

In terms of the foreground time it should be several hours.

Reply:
https://github.com/organicmaps/organicmaps/pull/9695#discussion_r1852288314

With this branch https://github.com/organicmaps/organicmaps/pull/9710
you can get US_* log entries (when the app starts) to check your usage. Several hours in foreground is way too much.

Reply:
The same 15min foreground time could differ wildly in terms of the exploration of app's features.

E.g. you're an experienced user and tend to be very efficient when you beta-test or use the app, but for a new user 15min could be just enough to download some maps and browse around a little. Or its just 15min of a drive (using the navigation feature) and not much else.

I think 15min won't be enough even for a weekend trip for new users.
Let's make it an hour at least? (could translate to somewhere between a week and a month of "real time", which looks reasonable)

It applies for new users only, of course.
For existing users there is no need to wait 15min even, e.g. make it 3min (as we're sure existing users have had the app installed at least 2 weeks before according to our update cycle).